start.py

```diff
@@ -0,0 +1,252 @@
+    conf = load_config.from_file(config_file, use_cli_input=False)
+
+    for _ in conf.startup_list:
+        if not check_if_tag_known(_.module):
+            continue
+
+        current_startup_list.append(
+            load_config.conf_to_named_tuple(data_types.StartupItem, _)
+        )
+
+    return current_startup_list
+
+
+def parse_startup_cli(modules: list, current_startup_list=None) -> list:
+    """
+    Convert module command line input into a list of StartupItem
+    """
+    if current_startup_list is None:
+        current_startup_list = []
+
+    for module in modules:
+        if not module:
+            continue
+
+        if not check_if_tag_known(module[0]):
+            continue
+
+        if len(module) > 1:
+            debug = bool(module[1] == "true" or module[1] == "True")
+
+        if len(module) == 1:  # Only modules specified
+            new_item = data_types.StartupItem(
+                module=module[0], debug=False, config_file=None, process_tag=module[0]
+            )
+        elif len(module) == 2:  # Debug specified
+            new_item = data_types.StartupItem(
+                module=module[0], debug=debug, config_file=None, process_tag=module[0]
+            )
+        elif len(module) == 3:  # Config file specified
+            new_item = data_types.StartupItem(
+                module=module[0],
+                debug=debug,
+                config_file=module[2],
+                process_tag=module[0],
+            )
+        elif len(module) == 4:  # Process name specified
+            new_item = data_types.StartupItem(
+                module=module[0],
+                debug=debug,
+                config_file=module[2],
+                process_tag=module[3],
+            )
+        else:
+            continue
+
+        current_startup_list.append(new_item)
+
+    return current_startup_list
+
+
+def launch_process(launch_item: data_types.StartupItem) -> subprocess.Popen:
+    """
+    Launch a subprocess from the provided StartupItem definition
+    """
+    if not check_if_tag_known(launch_item.module):
+        return None
+
+    argslist = ["python3", "-m"]
+    argslist.extend(launch_list[launch_item.module])
+
+    if launch_item.config_file is not None:
+        argslist.extend(["-c", launch_item.config_file])
+
+    if launch_item.debug:
+        argslist.extend(["-d"])
+
+    if launch_item.process_tag is not None:
+        argslist.extend(["-pt", launch_item.process_tag])
+
+    if launch_item.health_monitor:
+        argslist.extend(["-hm"])
+
+    return subprocess.Popen(
+        args=argslist, stderr=subprocess.PIPE, start_new_session=True
+    )
+
+
+def launch_processes(startup_items: list) -> list:
+    """
+    Launch all subprocesses in the specified list and collect their resulting Popen objects
+    """
+    process_list = []
+
+    for startup_item in startup_items:
+        new_process = launch_process(startup_item)
+
+        if new_process is None:
+            continue
```

Comment:
Silent failure of process launch much?

Reply:
Prints to stderr when None is received and added a TODO for when the logging system is implemented